vnext/docs/NativeModules.md

```diff
@@ -220,9 +250,11 @@ Then we define constants, and it's as easy as creating a public field and giving
 
 It's just as easy to add custom methods, by attributing a public method with `REACT_METHOD`. In FancyMath we have one method, `add`, which takes two doubles and returns their sum. Again, we've specified the optional `name` argument in the `REACT_METHOD` macro-attribute so in JS we call `add` instead of `Add`.
 
+To add custom events, we attribute a `std::function<void(double)>` delegate with `REACT_EVENT`, where the double represents the type of the event data. Now whenever we invoke the `AddEvent` delegate in our native code (as we do above), an event will be raised in JavaScript. As before, you could have optionally customized the name in JS like this: `REACT_EVENT(AddEvent, "addEvent")`.
```

Comment:
>JavaScript [](start = 263, length = 10)

does it mean this is the mechanism through which Javascript would get the return value of a native function? Is this the recommended way/are there other ways that aren't event based?

Reply:
Methods can provide one or two callbacks to return value or value+error.

---
In reply to: [340891998](https://github.com/microsoft/react-native-windows/pull/3561#discussion_r340891998) [](ancestors = 340891998)